### PR TITLE
!act Hide pipe and pipeToSelection from Java API.

### PR DESCRIPTION
**Pull request by He-Pin**
refs:https://github.com/akka/akka/issues/26003

### CODE REVIEW

File: akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala [hunk 1]
```diff
@@ -40,21 +39,17 @@ trait PipeToSupport {
...
 
   final class PipeableCompletionStage[T](val future: CompletionStage[T])(implicit executionContext: ExecutionContext) {
-    def pipeTo(recipient: ActorRef)(implicit sender: ActorRef = Actor.noSender): CompletionStage[T] = {
-      future whenComplete new BiConsumer[T, Throwable] {
-        override def accept(t: T, ex: Throwable): Unit = {
-          if (t != null) recipient ! t
-          if (ex != null) recipient ! Status.Failure(ex)
-        }
-      }
+    private def pipeTo(recipient: ActorRef)(implicit sender: ActorRef = Actor.noSender): CompletionStage[T] = {
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: keep these method private.

Comment (patriknw) [reply akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: I think this is an API breakage that we can't accept

Comment (patriknw) [reply akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: @hepin1989 do you understand my objection of why we can't change this? ok to close?

Comment (hepin1989) [reply akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: Yes, but could we defer this to 3.0?

Comment (hepin1989) [reply akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: > @hepin1989 do you understand my objection of why we can't change this? ok to close?

I am ok to close , but I would like to see this to be rescheduled to 3.0.

Comment (patriknw) [reply akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: Well, many changes would change if we would do a 3.0, but we don't have any such plans now.

Comment (hepin1989) [reply akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: I mean there is already an issue there, as this PR referenced. Could be added to there as a backlog,other than simply close it:)

**Comment by hepin1989** [comment]
Closed for Binary Compatibility, re-open with Akka 3.0?

File: akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala]: sure, did that in the ticket

**Pull request closed without merging**